File: app/src/main/java/org/baiyu/fuckshare/ClearCacheWorker.java
package org.baiyu.fuckshare;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import timber.log.Timber;

public class ClearCacheWorker extends Worker {
    public static final String id = "clearCache";
    private final Context context;

    public ClearCacheWorker(
            @NonNull Context context,
            @NonNull WorkerParameters params) {
        super(context, params);
        this.context = context;
    }

    @NonNull
    @Override
    public Result doWork() {
        final long timeDurationMillis = 1000 * 60 * 30; // 30 mins
        boolean result = Utils.clearCache(context, timeDurationMillis);
        Timber.d("Cache cleared with result: %b", result);
        return result ? Result.success() : Result.failure();
    }
}

File: app/src/main/java/org/baiyu/fuckshare/ClearCacheWorker.kt
package org.baiyu.fuckshare

import android.content.Context
import androidx.work.Worker
import androidx.work.WorkerParameters
import org.baiyu.fuckshare.utils.AppUtils
import timber.log.Timber

class ClearCacheWorker(
    private val context: Context,
    params: WorkerParameters
) : Worker(context, params) {
    override fun doWork(): Result {
        val timeDurationMillis = 1000L * 60L * 30L // 30 mins
        val result = AppUtils.clearCache(context, timeDurationMillis)
        Timber.d("Cache cleared with result: $result")
        return if (result) Result.success() else Result.failure()
    }

    companion object {
        const val id = "clearCache"
    }
}